### DISCUSSION TITLE
MuJoCo Menagerie

### DISCUSSION THREAD

**Discussion by yuvaltassa** [google-deepmind/mujoco#477] [Announcements]
We've released a curated repository of models: the **MuJoCo Menagerie**.

It lives in [github.com/deepmind/mujoco_menagerie](https://github.com/deepmind/mujoco_menagerie)

[![MuJoCo Menagerie](https://img.youtube.com/vi/3dlM1dvBtko/0.jpg)](https://www.youtube.com/watch?v=3dlM1dvBtko)

**Comment by Balint-H** [comment]
Is the Menagerie targeted for MuJoCo equivalents of robotic systems? Or will biomechanical or purely virtual models be supported as well? (by purely virtual I mean models that have no real life equivalents, such as classic RL environment scenes)

**Reply by yuvaltassa** [reply]
Biomechanical: absolutely!
Abstract models: probably not.
The reason is that we want models that have a clear notion of "good". For robots and biomechanical systems a good model is well-defined. For abstract models it is not.

Does that sound reasonable?

**Reply by Balint-H** [reply]
Thanks for supporting this, it sounds like an excellent resource and I really like the grading system included. 

I personally look forward the most to validated/recommended biomechanical human locomotion models of different levels of detail (e.g. some using muscle actuators, some not).

Potential use cases for abstract models I could see:

- Simpler analogues of real systems for cheaper/faster evaluation for prototyping and benchmarking (e.g. generic quadruped, generic biped), since these could still be rated for their sensibility of parameters and similiarity to real systems (with a grade likely not better than B). Some basic examples like this exist in the main repository, which might be more suitable for them. A couple more "official" examples of models for common generic tasks wouldn't be out of place.

- Use cases for graphics applications, for simulated characters and creatures (for procedural physics driven animation). Fidelity to reality is still important in these applications, but don't necessarily need a physical equivalent, or can sacrifice some aspect of modelling. Having validated baselines that produce high quality motion and environmental interaction could be useful in these settings too.

But I do see that it isn't as necessary to have fine-tuned models in the abstract cases. I suppose whether it makes sense to maintain and feature use cases like these would depend on the demand from the community, which might not be there.
